jsonpickle/unpickler.py

```diff
@@ -390,8 +390,8 @@ def register_classes(self, classes):
             for cls in classes:
                 self.register_classes(cls)
         elif isinstance(classes, dict):
-            for cls in classes.values():
-                self.register_classes(cls)
+            for cls in classes:
+                self._classes[cls] = classes[cls]
```

Comment:
We can save a dict lookup here by uisng `for cls, handler in classes.items(): self._classes[cls] = handler`.

Once it's written that way it becomes clearer that we don't even need to loop. It seems like we can use `.update()` instead, which is faster too -> `self._classes.update(classes)`

Make sure to run the tests. I just enabled the actions so we'll see the results from the CI tests shortly.

If we do take this then we'll at minimum need a test case to make sure that it's working as advertised. Similarly, that test will make sure I don't accidentally break this in the future =)

Thanks for the patch, this does sound like a useful feature at first glance.

Reply:
Once we see the test it might become clearer that the API needs adjustment. It's kinda interesting that we only looked at the keys.. it's almost like we left the door open to this but hadn't implemented this edge case.

I'm thinking that might actually want to use something like `for key, value in classes.items(): self._classes[util.importable_name(key)] = value` as the implementation.

Likewise, we wouldn't want users to have to stringify things themselves, as that's exposing a bit too much of our internal implementation to the outside world. It's be better to have the API take class objects in both the keys and values, i.e.

```python
jsonpickle.decode(json_str, classes={older.pkg.Class: newer.pkg.Class})
```

In other words, we'd expect the user to pass the class direclty in the key rather than the quoted importable name. That way there's less chance for errors.

Reply:
Last sug, I promise (for now) -- I do generally think that passing the classes directly is better.

But, I can imagine scenarios where `older.pkg.Class` is not importable, and we still want users to be able to remap the json data they have around without having to jump through too many hoops.

In that case, we should check if the key is a string and if it is then we can use it as-is. If it's not a string (e.g. it's a class/type object) then we'll call importable_name on it.

That seems like the best of both worlds. It'll be ergonomic for the common case and it'll also let users supply the string names directly.

Reply:
I think @davvid is correct with the suggestion of using ``self._classes.update(classes)``. His assessment that allowing both strings and class objects to work is ideal behavior, but I think that's work for another PR, as I generally take the view that it's best to push out bug fixes first and then add ideal behavior later, once there's enough developer time to go around.

Reply:
Agreed on both counts. In this PR, I've fixed the bug of not updating `self._classes` correctly, but it only supports string keys in the dict (concur with @Theelx).

I've made the change of using `.update()` directly.